record-editor.py

```diff
@@ -361,6 +358,12 @@ def create_widgets( self ):
         self.infoTakenLabel    = QLabel()
         self.infoTagsLabel     = QLabel()
 
+        # dock widget which will hold the selection layout once created
+        # in create_layout, for now it gets an empty widget.
+        self.selection_dock = QDockWidget()
```

Comment:
Can you educate me on the sizing strategy behind adding a dock?  From what I can tell it makes it easier to stretch the selection layout's size to be fully visible (and beyond, strangely) though don't understand if there are other benefits for resizing the PhotoRecordViewer window while maintaining preview aspect ratio, sensibly packed labels, etc.